patrickrottman/osrs-inventory-setups
# Added Bank Tag Layouts Support

# Bank Tag Layout Export Format Fix

## Changes
- Added `originalFormat` property to `LoadoutData` interface to store the original import format
- Updated `BankTagLayout` interface to include `originalFormat` field
- Modified `BankTagLayoutService` to preserve original format during parsing
- Updated `LoadoutUploaderDialogComponent` to store original format when creating layouts
- Modified `LoadoutModalComponent` to use original format when exporting if available
- Fixed type filter value mismatch between UI and service (changed from 'banktag' to 'banktaglayout')

## Why
Previously, when importing and then exporting bank tag layouts, the format would be regenerated which could cause issues with RuneLite plugin compatibility. This change ensures we preserve the exact format that was imported, maintaining compatibility with the RuneLite Bank Tag Layouts plugin.

## Testing
1. Import a bank tag layout using either format:
   - `banktaglayoutsplugin:name,itemId:position,...banktag:id1,id2,...`
   - `banktags,version,name,id1,id2,...`
2. Verify the layout displays correctly
3. Export the layout and verify the format matches the original import exactly
4. Verify filtering between Inventory Setups and Bank Tag Layouts works correctly

## Notes
- The original format is only stored for bank tag layouts, not inventory setups
- If no original format is available, falls back to generating a new format
